### ISSUE TITLE
Node connexion failure

### ISSUE THREAD

**Issue by Sami32** [BlakeKaufman/BlitzWallet#37]
Blitz wallet fail to connect to a node at the creation of a wallet (with or without VPN) using Android 7.0 MIUI 11 (Xiaomi).

![Screenshot_2024-11-06-13-47-39-493_com blitzwallet](https://github.com/user-attachments/assets/02862016-64e0-474a-9473-550e7b5fd023)

Edit:
Blitz wallet Android-v0.3.2-beta

**Comment by BlakeKaufman** [comment]
Could you try and connect on cell service and not wifi?

Also, was this the first time you loaded the wallet, if so, did you import a seed or use a seed generated by Blitz?

**Comment by Sami32** [comment]
No, because my cellular phone doesn't have a data plan so inserting my SIM card will not help.

I only used the creation mode, meaning generating a seed with Blitz.

**Comment by BlakeKaufman** [comment]
Was this the first time you have loaded the wallet? or has it loaded in the past and now is not loading?

**Comment by Sami32** [comment]
It was my first try ever of your App.

**Comment by BlakeKaufman** [comment]
I will have to check on this. I have sometimes noticed an issue connecting to the greenlight node on the initial load from some wifi services because it says "connection reset by peer". I will get back to you if I find anything.

**Comment by Sami32** [comment]
Thanks, no worry, nothing hurry.

I should add that Xiaomi is known to have some proprietary Java implementation that rise compatibility issues; perhaps that Xiaomi get a beter implementation after my Redmi Note 4?
https://www.browserstack.com/test-on-xiaomi

**Comment by Sami32** [comment]
I probably should have said that i disabled Google Play services.

**Comment by BlakeKaufman** [comment]
Here is the response I got from the Greenlight team:

I filled out the report based on this issue with my school wifi, but you are running into the same issue as well.

We've had this situation a couple of times before. If the connection to scheduler.gl.blckstrm.com is the only affected connection it is likely that your school is using DPI with a custom root certificate installed on the clients, that allows the firewall to decrypt the traffic between client and server. This strips not only the original server TLS envelope, but also the client authentication we use to identify the client.

We documented this in the FaQ here: https://github.com/Blockstream/greenlight/pull/546/files

Can you check with the network admins to see if they are doing DPI, and maybe ask what would be required for an exemption from it?

**Comment by Sami32** [comment]
I used it from home WiFi so if DPI is done it should be at my ISP level.

But then AQUA & GREEN or Wallby wallets shouldn't have been equally affected?
Because these wallets are working fine for me.

**Comment by BlakeKaufman** [comment]
It's not an issue with the liquid node its an issue with the greenlight node. Have you tried to log in again since it failed?

**Comment by Sami32** [comment]
Ok.

Yes, and i got the same error message.

**Comment by Sami32** [comment]
Using the latest pre-release version, now after creating a new wallet i get this new error instead (with or without VPN):

![Screenshot_2025-01-15-20-40-28-480_com blitzwallet](https://github.com/user-attachments/assets/d1b32be6-980f-4768-bcea-b48e6c1e95e9)

**Comment by BlakeKaufman** [comment]
Are you using this version?

https://github.com/BlakeKaufman/BlitzWallet/releases/tag/Android-v0.3.7-beta

**Comment by Sami32** [comment]
I use this one:

https://github.com/BlakeKaufman/BlitzWallet/releases/tag/Android-v0.3.7-preRelease

**Comment by BlakeKaufman** [comment]
try the actual release one that I sent

**Comment by Sami32** [comment]
If you meant the link that you provided above, then the error displayed is still the same.

**Comment by BlakeKaufman** [comment]
I just changed something on the backend, what does it do now?

**Comment by Sami32** [comment]
https://github.com/BlakeKaufman/BlitzWallet/releases/tag/Android-v0.3.7-beta

![Image](https://github.com/user-attachments/assets/5f760af0-8616-44c6-a585-f1bc2de06250)

**Comment by BlakeKaufman** [comment]
Try one more time, otherwise you might have to wait till the next update because I took this out.

**Comment by Sami32** [comment]
Same same.

Ok, no problem.

**Comment by BlakeKaufman** [comment]
Could you try this version?

 https://github.com/BlakeKaufman/BlitzWallet/releases/tag/Android-v0.3.8-preRelease

**Comment by Sami32** [comment]
Congratulation...you fixed it +1
